pkg/messages/treatyMessage.go

```diff
@@ -0,0 +1,116 @@
+package messages
+
+import (
+	"github.com/google/uuid"
+)
+
+type ConditionType int // type of condition
+type RequestType int
+type Op int
+
+const (
+	HP ConditionType = iota + 1
+	Floor
+	AvailableFood
+)
+
+const (
+	LeaveAmountFood RequestType = iota + 1
+	LeavePercentFood
+	Inform
+)
+
+const (
+	GT Op = iota + 1
+	GE
+	EQ
+	LE
+	LT
+)
+
+type Treaty struct {
+	condition      ConditionType
+	request        RequestType
+	conditionOp    Op
+	requestOp      Op
+	signatureCount int
+	duration       int
+	id             uuid.UUID
```

Comment:
Should these fields be exported or unexported?

At the very least, surely `signatureCount` should be exported? Or a `signTreaty()` function is needed?

Reply:
Unexported I think and Ill add `signTreaty()`